layouts/wallet.vue

```diff
@@ -13,40 +13,58 @@
 <script lang="ts">
 import { Vue, Component, Watch } from 'vue-property-decorator'
 import { namespace } from 'vuex-class'
+import logTrackerEvent, { setLoggerUser } from '~/utils/logger'
+
 
-const signerModule = namespace('signer')
 const walletModule = namespace('wallet')
 
 @Component
 export default class WalletLayout extends Vue {
-  @walletModule.State('isShowConnectDialog') isShowConnectWalletDialog!: boolean
-  @walletModule.Action('toggleConnectDialog') toggleConnectWalletDialog!: (
-    isShow: boolean
-  ) => void
+  @walletModule.Getter('getWalletAddress') walletAddress!: string
+  @walletModule.Action('openConnectWalletModal') openConnectWalletModal!: (params: { language: string }) => Promise<any>
+  @walletModule.Action('initWallet') initWallet!: (params: { method: any, accounts: any, offlineSigner?: any }) => Promise<any>
+  @walletModule.Action('restoreSession') restoreSession!: () => Promise<any>
 
   @walletModule.Action toggleAlert!: (
     isShow: boolean,
     error: string,
     severity: string
   ) => void
 
-  @signerModule.Getter('getAddress') currentAddress!: string
-
   get isHideFooter() {
     return this.$route.path.includes('/nft/purchase/') || this.$route.query.layout === 'popup';
   }
 
-  mounted() {
-    if (!this.currentAddress) {
-      this.toggleConnectWalletDialog(true)
+  async mounted() {
+    await this.restoreSession()
```

Comment:
Isn't `restoreSession()` already done in `<RootLayout/>`?

Reply:
Yes, but there is a time lag when refreshing the [mint /] or [publishing /] (RootLayout + WalletLayout) screens in place, causing the background to enter the RootLayout screen while the WalletLayout is still requesting login.

Reply:
I think `restoreSession` should set a state e.g. `isRestoringSession`??, so you don't double restoring the session